tests/scripts/ping_reviewers.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import argparse
+import re
+import datetime
+import json
+from typing import Dict, Any, List
+
+from git_utils import git, GitHubRepo, parse_remote
+
+
+def prs_query(user: str, repo: str, cursor: str = None):
+    after = ""
+    if cursor is not None:
+        after = f', before:"{cursor}"'
+    return f"""
+        {{
+    repository(name: "{repo}", owner: "{user}") {{
+        pullRequests(states: [OPEN], last: 10{after}) {{
+        edges {{
+            cursor
+        }}
+        nodes {{
+            number
+            url
+            body
+            isDraft
+            author {{
+                login
+            }}
+            reviews(last:100) {{
+                nodes {{
+                    author {{ login }}
+                    comments(last:100) {{
+                        nodes {{
+                            updatedAt
+                            bodyText
+                        }}
+                    }}
+                }}
+            }}
+            publishedAt
+            comments(last:100) {{
+                nodes {{
+                    authorAssociation
+                    bodyText
+                    updatedAt
+                    author {{
+                        login
+                    }}
+                }}
+            }}
+        }}
+        }}
+    }}
+    }}
+    """
+
+
+def find_reviewers(body: str) -> List[str]:
```

Comment:
is there a way to query the "Review requests" from GH instead? after we turn off CODEOWNERS this field will become more useful.

Reply:
We can add that functionality later if need be, but most PRs people are going to be cc-ing people anyways due to the other bot. The main reason I didn't add it here is that non-committers can't be added to the reviewers list until they've actually left a review (which this does query for)